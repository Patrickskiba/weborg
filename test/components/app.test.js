import React from 'react'
import { render, cleanup, fireEvent } from 'react-testing-library'
import 'jest-dom/extend-expect'
import jsdom from 'jsdom'


jest.mock('dropbox')

jest.mock('../../src/components/dropbox-files', () => ({setText}) => { 
    const testText = 
`
* Great Unix Tools
** rsync
   Copy a file with a progress bar
   sudo rsync --info=progress2 source dest
** du - disk usage
   du -sh file_path
   -s : summarized
   -h : human readable
** pacman
   search pacman
   - sudo pacman -Ss package_name
`
    setText(testText)

    return <div></div>
})

describe('app tests', () => {
    it('renders the component with the correct headlines and collapses the topmost headline', async () => {
        const mockedDropbox = require('dropbox')
        const mockedDropboxFiles = require('../../src/components/dropbox-files')

        window.location.hash = '#access_token=test'

        const App = require('../../src/components/app').default
        const { getAllByText, getAllByTestId, container } = render(<App />)

        const headlines = getAllByTestId('headline')

        expect(headlines[0]).toHaveTextContent("Great Unix Tools")
        expect(headlines[1]).toHaveTextContent("rsync Copy a file with a progress bar sudo rsync --info=progress2 source dest -")
        expect(headlines[2]).toHaveTextContent("du - disk usage du -sh file_path -s : summarized -h : human readable -")
        expect(headlines[3]).toHaveTextContent("pacman search pacman - sudo pacman -Ss package_name -")

        fireEvent.click(getAllByText('-')[3], { button: 1 } )

        const condensedHeadlines = getAllByTestId('headline')

        expect(condensedHeadlines[0]).toHaveTextContent("Great Unix Tools")
        expect(condensedHeadlines[1]).toEqual(undefined)
        expect(condensedHeadlines[2]).toEqual(undefined)
        expect(condensedHeadlines[3]).toEqual(undefined)

        expect(container).toMatchSnapshot()

        fireEvent.click(getAllByText('+')[0], { button: 1 } )

        const uncondensedHeadlines = getAllByTestId('headline')

        expect(uncondensedHeadlines[0]).toHaveTextContent("Great Unix Tools")
        expect(uncondensedHeadlines[1]).toHaveTextContent("rsync Copy a file with a progress bar sudo rsync --info=progress2 source dest -")
        expect(uncondensedHeadlines[2]).toHaveTextContent("du - disk usage du -sh file_path -s : summarized -h : human readable -")
        expect(uncondensedHeadlines[3]).toHaveTextContent("pacman search pacman - sudo pacman -Ss package_name -")

    })
})
