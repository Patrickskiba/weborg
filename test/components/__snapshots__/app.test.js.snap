// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app tests renders the component with the correct headlines and collapses the topmost headline 1`] = `
<div>
  <div
    class="sc-htoDjs bPqIrB"
  >
    <div
      class="sc-dnqmqq cZTbth"
    >
      <div>
        <img
          src="dropbox.svg"
          style="width: 25px;"
        />
      </div>
      <div />
    </div>
    <div
      class="sc-iwsKbI bPAHQp"
    >
      <button>
        Hide/Show
      </button>
      <div
        class="sc-gzVnrw eJhfXY"
      >
        <span>
          <span />
           
        </span>
      </div>
      <div
        class="sc-bdVaJa kDGlbf"
        data-testid="headline"
      >
        <div
          class="sc-htpNat iqqKqB"
        >
          <div
            class="sc-ifAKCX dJvHBo"
          />
          <div
            class="sc-bxivhb bRilrT"
          >
            <div
              style="margin-right: 5px;"
            >
              <svg
                height="16px"
                width="16px"
              >
                <circle
                  cx="50%"
                  cy="50%"
                  fill="#e1e1e1"
                  r="8px"
                />
                <circle
                  cx="50%"
                  cy="50%"
                  fill="#616161"
                  r="4px"
                />
              </svg>
            </div>
          </div>
          <div
            class="sc-EHOje cxKCIN"
          >
            <div>
              <span
                style="color: green; font-weight: 600;"
              >
                 
                 
              </span>
              <span>
                <span>
                  Great Unix Tools
                </span>
                 
              </span>
            </div>
          </div>
          <div
            class="sc-bZQynM gDQOZZ"
          >
            <div>
              +
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
